### PR TITLE
feat: create-oryx utility

### PR DESCRIPTION
**Pull request by dunqan**
Support for `npm create oryx`. 
Precompiled wrapper for Oryx CLI, allowing to easily setup an Oryx application. 

Closes [HRZ-86543](https://spryker.atlassian.net/browse/HRZ-86543)

[HRZ-86543]: https://spryker.atlassian.net/browse/HRZ-86543?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/template/cli/package.json [hunk 1]
```diff
@@ -10,14 +10,15 @@
   "bin": {
     "oryx": "./bin/cli.js"
   },
-  "dependencies": {
+  "devDependencies": {
```
Comment (tolerants) [inline libs/template/cli/package.json]: Why do you think these are dev dependencies?

File: libs/template/cli/src/commands/create.ts [hunk 2]
```diff
@@ -79,6 +80,7 @@ Possible values: ${Object.values(OryxPreset).join(', ')}`
   }
 
   async createApp(options: CreateAppOptions = {}): Promise<void> {
+    console.log(``);
```
Comment (tolerants) [inline libs/template/cli/src/commands/create.ts:68]: Leftover?

File: libs/template/cli/src/commands/create.ts [hunk 3]
```diff
@@ -123,77 +134,58 @@ Please make sure to not use an existing directory name.`
...
-        message: `What is the name of your app? ${c.dim(`[--name, -n]`)}`,
-        validate: (value) => {
-          if (!value.trim()) {
-            return 'Please enter a name';
-          }
-          return undefined;
-        },
+        message: `What is the name of your app?`,
+        placeholder: 'oryx-app',
+        defaultValue: 'oryx-app',
```
Comment (tolerants) [inline libs/template/cli/src/commands/create.ts:163]: Maybe `initialValue` will be better?

File: libs/template/cli/src/commands/create.ts [hunk 4]
```diff
@@ -239,13 +231,22 @@ interface CreateAppConfig extends Required<CreateAppOptions> {
...
   B2C = 'b2c',
+  B2B = 'b2b',
   Fulfillment = 'fulfillment',
 }
 
 export enum OryxTemplateRef {
   Latest = 'latest',
 }
+
+export enum OryxOption {
```
Comment (tolerants) [inline libs/template/cli/src/commands/create.ts:192]: Probably we can remove it as long as we are not using it yet.

File: libs/template/cli/src/commands/create.ts [hunk 5]
```diff
@@ -239,13 +231,22 @@ interface CreateAppConfig extends Required<CreateAppOptions> {
 export interface CreateAppOptions {
   name?: string;
   preset?: OryxPreset;
+  options?: OryxOption[];
 }
 
 export enum OryxPreset {
```
Comment (tolerants) [inline libs/template/cli/src/commands/create.ts]: Probably we can remove it as long as we are not using it yet.

**Review by tolerants — CHANGES_REQUESTED** [review]

File: libs/template/cli/src/commands/create.ts (hunk 2 above)
Comment (dunqan) [reply libs/template/cli/src/commands/create.ts:68]: nope, empty line

File: libs/template/cli/src/commands/create.ts (hunk 3 above)
Comment (dunqan) [reply libs/template/cli/src/commands/create.ts:163]: initial you'll have to edit if you want to use different name, while default will be used if no name is provided, or you can just write your own

File: libs/template/cli/src/commands/create.ts (hunk 5 above)
Comment (dunqan) [reply libs/template/cli/src/commands/create.ts]: we can, up to you :)

File: libs/template/cli/package.json (hunk 1 above)
Comment (dunqan) [reply libs/template/cli/package.json]: they are not yet, but I think we should build cli bin to make it work without all the oryx deps, like `create-oryx`. Then they will be real devDeps.

**Comment by tolerants** [comment]
As long as we are not going to release it, probably it's a `chore`.

**Review by tolerants — DISMISSED** [review]

**Review by supproduction — APPROVED** [review]

**Pull request merged**